proposals/lip-0044.md

```diff
@@ -534,7 +533,7 @@ def getValidatorParams() -> ValidatorParams:
 
 #### getGeneratorsBetweenTimestamps
 
-This function returns the addresses of the generators active between the two input timestamps and the number of block slots assigned to them. The slots corresponding to the input timestamps are NOT counted, only the slots in between. Notice that for all counted slots, the current `generatorList` must be the used generator list. Otherwise, the behaviour is undefined. It’s the calling module’s responsibility to ensure that.
+This function returns the addresses of the generators active between the two input timestamps and the number of block slots assigned to them. The slots corresponding to the input timestamps are NOT counted, only the slots in between. Notice that for all counted slots, the current value of `validatorParamsStore.validators` must be the used generator list. Otherwise, the behaviour is undefined. It’s the calling module’s responsibility to ensure that.
```

Comment:
I find this paragraph a bit confusing. Do we want to say that a module (e.g.,PoA or PoS) calling this function should ensure that the validators in the validator params substore are aligned to its current validators? If yes I would explicitly state it like that; currently it does not seem clear to me what "must be the used generator list" means and what we mean by undefined behavior.

Reply:
Suggestion:

> Notice that this function assumes that `validatorParamsStore.validators` is the list of generators used in the engine to assign validators to block slots in a round-robin fashion (see function getGeneratorAtTimestamp in [LIP 0058](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0058.md#getgeneratorattimestamp)).

Reply:
I updated the description. The point is that the input timestamps can, for instance, NOT be chosen that they refer to some rounds in the past where older validator set was used. The function only works for time stamps since the latest change of  `validatorParamsStore.validators`.